src/node/mod.rs

```diff
@@ -0,0 +1,1241 @@
+#[cfg(feature = "mock_base")]
+impl Node {
+    /// Returns whether the node is approved member of a section.
+    pub fn is_approved(&self) -> bool {
+        self.stage.approved().is_some()
+    }
+
+    /// Returns whether the node is Elder.
+    pub fn is_elder(&self) -> bool {
+        self.chain()
+            .map(|chain| chain.is_self_elder())
+            .unwrap_or(false)
+    }
+
+    /// Indicates if there are any pending observations in the parsec object
+    pub fn has_unpolled_observations(&self) -> bool {
+        self.stage
+            .approved()
+            .map(|stage| stage.parsec_map.has_unpolled_observations())
+            .unwrap_or(false)
+    }
+
+    /// Indicates if there are any pending observations in the parsec object
+    pub fn unpolled_observations_string(&self) -> String {
+        self.stage
+            .approved()
+            .map(|stage| stage.parsec_map.unpolled_observations_string())
+            .unwrap_or_else(String::new)
+    }
+
+    /// Returns whether the given peer is an elder of our section.
+    pub fn is_peer_our_elder(&self, pub_id: &PublicId) -> bool {
+        self.stage
+            .approved()
+            .map(|stage| stage.chain.is_peer_our_elder(pub_id))
+            .unwrap_or(false)
+    }
+
+    /// Send a message to the given targets using the given delivery group size.
+    pub fn send_message_to_targets(
+        &mut self,
+        dst_targets: &[SocketAddr],
+        dg_size: usize,
+        message: Message,
+    ) -> Result<(), RoutingError> {
+        let message = message.to_bytes()?;
+        self.core
+            .send_message_to_targets(dst_targets, dg_size, message);
+        Ok(())
+    }
+
+    /// Returns the version of the latest Parsec instance of this node.
+    pub fn parsec_last_version(&self) -> u64 {
+        self.stage
+            .approved()
+            .map(|stage| stage.parsec_map.last_version())
+            .unwrap_or(0)
+    }
+
+    /// Checks whether the given location represents self.
+    pub fn in_src_location(&self, src: &SrcLocation) -> bool {
+        self.stage
+            .approved()
+            .map(|stage| stage.chain.in_src_location(src))
+            .unwrap_or(false)
+    }
+
+    /// Returns the prefixes of all our neighbours
+    pub fn neighbour_prefixes(&self) -> BTreeSet<Prefix<XorName>> {
+        self.chain()
+            .map(|chain| {
+                chain
+                    .neighbour_infos()
+                    .map(|info| info.prefix())
+                    .copied()
+                    .collect()
+            })
+            .unwrap_or_default()
+    }
+
+    /// Returns the prefixes of all sections known to us
+    pub fn prefixes(&self) -> BTreeSet<Prefix<XorName>> {
+        self.chain().map(Chain::prefixes).unwrap_or_default()
+    }
+
+    /// Returns the elder info version of a section with the given prefix.
+    /// Prefix must be either our prefix or of one of our neighbours.
+    /// Returns 0 otherwise.
```

Comment:
can this default `0` being a version of the gensis section ?

Reply:
It can. It should probably return `Option` to disambiguate this case. Will change it.